src/entity/Entity.js

```diff
@@ -94,4 +94,128 @@ export default class Entity {
   toString() {
     return JSON.stringify(this.toObject());
   }
+  /**
+   * generate a setter function
+   * @param {string} field - the field that you want to set
+   * @param {any} param - the variable that you to be setted
+   * @param {Array<string|types>} expectedTypes - the list of the authorized types for the param
+   * @return {function} - the setter function corresponding to the arguments
+   */
+   generateSetterFunction(field, param, expectedTypes = []) {
+    // at the end if checked = false we should throw an error because
+    // the param has not the expected type
+    let checked = false;
+    if (expectedTypes.length === 0) { // we check if the array is not empty
+      throw new Error('there must be at least one expected type');
+    }
+    expectedTypes.forEach(
+      (type) => {
+        // we check if the type is a primitive type : ['string','number','boolean',...]
```

Comment:
This part seems to be the same for the 3 functions, you should move it to a separated function